pkg/cmd/pr/merge/merge.go

```diff
@@ -224,7 +227,7 @@ func (m *mergeContext) warnIfDiverged() {
 		return
 	}
 
-	localBranchLastCommit, err := git.LastCommit()
+	localBranchLastCommit, err := m.opts.GitClient.LastCommit(context.Background())
```

Comment:
If you are repeating `context.Background()` within the same scope a lot, you could assign it to a variable `ctx := context.Background()` and then just pass `ctx` around. That will make it easier in the future to change `ctx` to be obtained from Cobra (or other sources) instead of using `context.Background()` .